File: cpp/Medium/2149_rearrange_array_elements_by_sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int>pos;
        vector<int>neg;
        vector<int>arr;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] >= 0) pos.push_back(nums[i]);
            if(nums[i] < 0) neg.push_back(nums[i]);
        }
        for(int i = 0; i < nums.size()/2; i++){
            arr.push_back(pos[i]);
            arr.push_back(neg[i]);
        }
        return arr;
    }
};

File: python/medium/2149_rearrange_array_elements_by_sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        positive = []
        negative = []
        ans = []
        for n in nums:
            if n > 0:
                positive.append(n)
            else:
                negative.append(n)
        
        for i in range(len(positive)):
            ans.append(positive[i])
            ans.append(negative[i])
        return ans
